src/platform/datapath_raw_socket.c

```diff
@@ -342,12 +342,77 @@ CxPlatDpRawCalculateHeaderBackFill(
     return HeaderBackFill;
 }
 
+_IRQL_requires_max_(DISPATCH_LEVEL)
+uint16_t
+CxPlatFramingChecksum(
```

Comment:
On the question of perf, how important is SW xsum? Should we practically make it a requirement that the HW support it or what? If we believe SW xsum is important, I don't believe we should "roll our own", as seen by all your other questions below.

Reply:
Assuming we have to do SW xsum, if we don't roll our own, what do we use? We have a very optimized tcpxsum implementation windows but I doubt it can be exposed to QUIC.

Reply:
Yeah this is the top level question I have - if checksum perf is a small fraction of encryption perf, then I see no problem and feel free to ignore the rest of my questions. Re: what to use, if the Windows tcpxsum routine turns out to be much faster *and* the perf impact is substantial, then we should consider porting it (or a similar idea) to QUIC.

Reply:
I think it's best to go forward with something and then iterate on the perf. Let's merge it, see how things are and go from there.

Reply:
Sounds good to me.

Reply:
Yeah, I didn't mean to just make one PR and we have the perfect xsum implementation. I am just getting it to work in sense of being able to interop with real clients and servers and then we party on how we can optimize it.

Reply:
FWIW, I've often seen the number "+20%" thrown around with respect to the cost of calculating checksums in software - but that's assuming a plaintext protocol.